scripts/extensions/sams/src/store/assets/actions.ts

```diff
@@ -253,16 +253,32 @@ export function deleteAsset(asset: IAssetItem): IThunkAction<void> {
     };
 }
 
-export function deleteMultipleAssets(): (dispatch: any, getState: any) => void {
+export function deleteAssets(asset?: IAssetItem): IThunkAction<void> {
     return (dispatch, getState) => {
-        const selectedAssets = getSelectedAssetItems(getState());
-
-        selectedAssets.map((selectedAsset) =>
-            dispatch(deleteAsset(selectedAsset))
-                .then(() => {
-                    dispatch(closeMultiActionBar());
-                }),
-        );
+        if (asset !== undefined) {
+            return openDeleteConfirmationModal(asset.name)
+                .then((response: boolean) => {
+                    if (response === true) {
+                        dispatch(deleteAsset(asset));
+                    }
+                });
+        } else {
+            const selectedAssets = getSelectedAssetItems(getState());
+            const assetName = (selectedAssets.length === 1) ? selectedAssets[0].name : undefined;
+
+            return openDeleteConfirmationModal(assetName, selectedAssets.length)
+                .then((response: boolean) => {
+                    if (response === true) {
+                        selectedAssets.map((selectedAsset) =>
+                            dispatch(deleteAsset(selectedAsset))
+                                .then(() => {
+                                    dispatch(closeMultiActionBar());
```

Comment:
Won't this dispatch the `closeMultiActionBar` for every Asset deleted? If that is the case, It might be better to run this once after all Promises have returned. i.e.
```js
return Promise.all(
    selectedAssets.map(
        (selectedAsset) => dispatch(deleteAsset(selectedAsset))
    )
).finally(() => {
    dispatch(closeMultiActionBar())
});
```